sdk/protocol2.go

```diff
@@ -0,0 +1,173 @@
+package sdk
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/cache"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/infra-integrations-sdk/metric"
+)
+
+// Entity is the producer of the data. Entity could be a host, a container, a pod, or whatever unit of meaning.
+type Entity struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// EntityData defines all the data related to a particular event from an entity.
+type EntityData struct {
+	Entity    Entity             `json:"entity"`
+	Metrics   []metric.MetricSet `json:"metrics"`
+	Inventory Inventory          `json:"inventory"`
+	Events    []Event            `json:"events"`
+}
+
+// EntitiesData is a collection of EntityData.
+type EntitiesData []*EntityData
+
+// Has tells you if that Entity is already part of the data.
+func (d EntitiesData) Has(entity Entity) bool {
+	for _, e := range d {
+		if e.Entity.Name == entity.Name && e.Entity.Type == entity.Type {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IntegrationProtocol2 defines the format of the output JSON that integrations will return for protocol 2.
+type IntegrationProtocol2 struct {
+	sync.Mutex
+	Name               string       `json:"name"`
+	ProtocolVersion    string       `json:"protocol_version"`
+	IntegrationVersion string       `json:"integration_version"`
+	Data               EntitiesData `json:"data"`
+	prettyOutput       bool
+	writer             io.Writer
```

Comment:
What can be the use case of it?

Reply:
It's being used in the `protocol_test.go`.